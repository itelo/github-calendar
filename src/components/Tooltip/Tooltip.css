.tooltip {
  position: relative;
}

.tooltip-trigger {
  text-decoration: underline;
}

.tooltip-bubble {
  min-width: 230px;
  max-width: 230px;
  position: absolute;
  z-index: 10;
}
.tooltip-bubble::after {
  content: "";
  position: absolute;
}

.tooltip-top {
  bottom: 100%;
  left: 50%;
  padding-bottom: 5px;
  transform: translateX(-50%);
}
.tooltip-top::after {
  border-left: 5px solid transparent;
  border-right: 5px solid transparent;
  border-top: 5px solid rgba(0, 0, 0, 0.7);
  bottom: 0;
  left: 50%;
  transform: translateX(-50%);
}

.tooltip-message {
  background: rgba(0, 0, 0, 0.7);
  border-radius: 3px;
  padding-top: 8px;
  padding-bottom: 8px;
  text-align: center;
}
