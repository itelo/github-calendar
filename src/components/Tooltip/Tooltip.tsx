import React from "react";
import "./Tooltip.css";

export default class Tooltip extends React.Component<any, any> {
  constructor(props: any) {
    super(props);

    this.state = {
      displayTooltip: false
    };
    this.hideTooltip = this.hideTooltip.bind(this);
    this.showTooltip = this.showTooltip.bind(this);
  }

  hideTooltip() {
    this.setState({ displayTooltip: false });
  }
  showTooltip() {
    this.setState({ displayTooltip: true });
  }

  render() {
    let message = this.props.message;
    let position = this.props.position;
    return (
      <div
        data-testid="tooltip"
        className="tooltip"
        onMouseLeave={this.hideTooltip}
      >
        {this.state.displayTooltip && (
          <div
            data-testid="tooltip-bubble"
            className={`tooltip-bubble tooltip-${position}`}
            onMouseOver={this.hideTooltip}
          >
            <div data-testid="tooltip-message" className="tooltip-message">
              {message}
            </div>
          </div>
        )}
        <div
          data-testid="tooltip-trigger"
          className="tooltip-trigger"
          onMouseOver={this.showTooltip}
        >
          {this.props.children}
        </div>
      </div>
    );
  }
}
