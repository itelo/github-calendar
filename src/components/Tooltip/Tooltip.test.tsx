import React from "react";
import { render, fireEvent } from "react-testing-library";
import "react-testing-library/cleanup-after-each";
import Tooltip from "./Tooltip";
import "jest-dom/extend-expect";

describe("<Tooltip />", () => {
  test("snapshot hide tooltip", () => {
    const { container } = render(
      <Tooltip message="MessageText" position="top">
        <div />
      </Tooltip>
    );

    expect(container).toMatchSnapshot();
  });

  test("snapshot show tooltip", () => {
    const { container, getByTestId } = render(
      <Tooltip message="MessageText" position="top">
        <div />
      </Tooltip>
    );

    const tooltipTrigger = getByTestId("tooltip-trigger");

    fireEvent.mouseOver(tooltipTrigger);

    expect(container).toMatchSnapshot();
  });

  test("test show hide tooltip", () => {
    const { getByTestId, queryByTestId } = render(
      <Tooltip message="MessageText" position="top">
        <div />
      </Tooltip>
    );

    const tooltip = getByTestId("tooltip");
    const tooltipTrigger = getByTestId("tooltip-trigger");

    expect(tooltip).not.toContainElement(queryByTestId("tooltip-message"));

    fireEvent.mouseOver(tooltipTrigger);

    expect(queryByTestId("tooltip-message")).toContainHTML("MessageText");

    fireEvent.mouseLeave(tooltip);
  });
});
