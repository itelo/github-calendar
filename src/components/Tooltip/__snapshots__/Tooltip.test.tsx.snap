// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Tooltip /> snapshot hide tooltip 1`] = `
<div>
  <div
    class="tooltip"
    data-testid="tooltip"
  >
    <div
      class="tooltip-trigger"
      data-testid="tooltip-trigger"
    >
      <div />
    </div>
  </div>
</div>
`;

exports[`<Tooltip /> snapshot show tooltip 1`] = `
<div>
  <div
    class="tooltip"
    data-testid="tooltip"
  >
    <div
      class="tooltip-bubble tooltip-top"
      data-testid="tooltip-bubble"
    >
      <div
        class="tooltip-message"
        data-testid="tooltip-message"
      >
        MessageText
      </div>
    </div>
    <div
      class="tooltip-trigger"
      data-testid="tooltip-trigger"
    >
      <div />
    </div>
  </div>
</div>
`;
